core/server/services/rss/generate-feed.js

```diff
@@ -21,35 +22,31 @@ generateTags = function generateTags(data) {
 
 generateItem = function generateItem(post, siteUrl, secure) {
     var itemUrl = urlService.getUrlByResourceId(post.id, {secure: secure, absolute: true}),
-        htmlContent = urlService.utils.makeAbsoluteUrls(post.html, siteUrl, itemUrl),
+        htmlContent = cheerio.load(post.html, {decodeEntities: false}),
         item = {
             title: post.title,
             // @TODO: DRY this up with data/meta/index & other excerpt code
-            description: post.custom_excerpt || post.meta_description || downsize(htmlContent.html(), {words: 50}),
+            description: post.custom_excerpt || post.meta_description || downsize(post.html, {words: 50}),
             guid: post.id,
             url: itemUrl,
             date: post.published_at,
             categories: generateTags(post),
             author: post.primary_author ? post.primary_author.name : null,
             custom_elements: []
-        },
-        imageUrl;
+        };
 
     if (post.feature_image) {
-        imageUrl = urlService.utils.urlFor('image', {image: post.feature_image, secure: secure}, true);
```

Comment:
We have talked about the `secure` flag behaviour. So the model layer doesn't have access to `req.secure`, which tells Ghost if the web server has setup https. Have you decided to ignore that for now? 🤔 Because if you set your blog url to http, you won't get https urls.

Reply:
We spoke about this after lunch and decided that we would continue with how this is. If the blog is setup for https correctly they'll get https urls, and it should also help move toward a new world of only https